js/src/data/types.js

```diff
@@ -9,7 +9,33 @@
  * @property {string[]} square_marketing_images The URLs of the square images.
  * @property {string[]} logo The URLs of the logo images.
  * @property {string[]} display_url_path The path part of the display URL on the ad.
- * @property {string | null} call_to_action The call-to-action text on the ad to let users know what the ad will get them to do. `null` if not selected.
+ * @property {string | null} call_to_action_selection The call-to-action text on the ad to let users know what the ad will get them to do. `null` if not selected.
+ */
+
+/**
+ * @typedef {Object} AssetEntity
+ * @property {number} id The ID of the asset.
+ * @property {string} content The content of the asset.
+ */
+
+/**
+ * @typedef {Object} AssetsDictionary
```

Comment:
💅  As an Idea we can follow the naming in the backend. `Asset`, `AssetGroup` and `AssetGroupAssets`

Reply:
Originally I had tried to use the naming of `Asset` and `AssetGroup`, but considering that there are many variables, docs or filenames in the frontend that use/include the same naming, they imply more abstract structures or different things like components. About `AssetGroupAsset`, it's the name that describes the link between an `Asset` unit and an `AssetGroup` unit at the database layer of Goodle Ads. But there is no such concept of data link in the frontend, because when getting the data from backend API, `Asset` and `AssetGroup` are already linked together as a result.

The purpose here is to explicitly indicate that a structure belongs to the data obtained by the backend API, so I finally added `Entity` and `Dictionary` to distinguish them.

Reply:
So its not this estructure the one we get on `get_assets_by_asset_group_ids` when we call `ads/campaigns/asset-groups`? 

Another way of seeing it. Having that consistency between BE and FE could help to understand connections between both sides of the implementation. As always naming is a very subjective thing.

Reply:
> So its not this estructure the one we get on `get_assets_by_asset_group_ids` when we call `ads/campaigns/asset-groups`?

It is, but the type relies on the backend APIs rather than the actual or internal implementation behind the APIs. When possible, it would be nice to have the same naming between backend and frontend. However, the naming styles (`final_url` and `finalUrl`) between them are different by default, and the targets and layers of abstraction are different as well. Therefore, it's usually not that easier to make them consistent.

Here it uses `AssetsDictionary` instead of other names I thought of while developing, e.g. `Assets`, `AssetsData`, `AssetsObject`, `AssetEntities`, `AssestEntity`, because I thought it's the least confusing one with the other names.